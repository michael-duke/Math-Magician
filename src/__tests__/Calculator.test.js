const { render, screen, fireEvent } = require('@testing-library/react');
require('@testing-library/jest-dom');
const Calculator = require('../components/Calculator').default;

describe('Tests for Calculator component', () => {
  it('Renders Calculator page', () => {
    const calculator = render(<Calculator />);
    expect(calculator).toMatchSnapshot();
  });

  it('Testing if the heading \'Let\'s do some math!\' is on the Calculator Page', () => {
    render(<Calculator />);
    const calculatorHeading = screen.getByText(/Let's do some math!/i);
    expect(calculatorHeading).toBeInTheDocument();
  });

  it('Simulating simple user interaction 121 is printed on dislpay', () => {
    render(<Calculator />);
    const displayNumber = document.getElementById('display');
    const key1 = screen.getByText('1');
    const key2 = screen.getByText('2');
    fireEvent.click(key1);
    fireEvent.click(key2);
    fireEvent.click(key1);
    expect(displayNumber.textContent).toBe('121');
  });

  it('Simulating user interaction 4*7 = 28', () => {
    render(<Calculator />);

    const displayNumber = document.getElementById('display');
    const key4 = screen.getByText('4');
    const key7 = screen.getByText('7');
    const keyMult = screen.getByText('*');
    const keyEquals = screen.getByText('=');

    fireEvent.click(key4);
    fireEvent.click(keyMult);
    fireEvent.click(key7);
    fireEvent.click(keyEquals);

    expect(displayNumber.textContent).toBe('28');
  });
});
