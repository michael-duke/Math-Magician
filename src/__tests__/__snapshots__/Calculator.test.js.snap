// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Calculator component Renders Calculator page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center gap-10 sm:flex-col sm:items-center mt-7"
      >
        <h2
          class="text-2xl font-semibold text-magic-blue self-start sm:self-center"
        >
          Let's do some Math!
        </h2>
        <section
          class="Calculator border-2 border-magic-blue rounded-lg overflow-hidden shadow-calcShadow"
        >
          <div
            class="grid grid-rows-6 grid-cols-4 gap-2"
          >
            <div
              class="col-span-4 bg-gray-400 text-white text-zl flex items-center justify-end m-[-4px]"
              id="display"
            >
              0
            </div>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              AC
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              +/-
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              %
            </button>
            <button
              class="bg-magic-blue/80 text-white"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              รท
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              7
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              8
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              9
            </button>
            <button
              class="bg-magic-blue/80 text-white"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              *
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              4
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              5
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              6
            </button>
            <button
              class="bg-magic-blue/80 text-white"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              -
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              1
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              2
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              3
            </button>
            <button
              class="bg-magic-blue/80 text-white"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              +
            </button>
            <button
              class="col-span-2 bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              0
            </button>
            <button
              class="bg-slate-200"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              .
            </button>
            <button
              class="bg-magic-blue/80 text-white"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              =
            </button>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center gap-10 sm:flex-col sm:items-center mt-7"
    >
      <h2
        class="text-2xl font-semibold text-magic-blue self-start sm:self-center"
      >
        Let's do some Math!
      </h2>
      <section
        class="Calculator border-2 border-magic-blue rounded-lg overflow-hidden shadow-calcShadow"
      >
        <div
          class="grid grid-rows-6 grid-cols-4 gap-2"
        >
          <div
            class="col-span-4 bg-gray-400 text-white text-zl flex items-center justify-end m-[-4px]"
            id="display"
          >
            0
          </div>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            AC
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            +/-
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            %
          </button>
          <button
            class="bg-magic-blue/80 text-white"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            รท
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            7
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            8
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            9
          </button>
          <button
            class="bg-magic-blue/80 text-white"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            *
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            4
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            5
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            6
          </button>
          <button
            class="bg-magic-blue/80 text-white"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            -
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            1
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            2
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            3
          </button>
          <button
            class="bg-magic-blue/80 text-white"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            +
          </button>
          <button
            class="col-span-2 bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            0
          </button>
          <button
            class="bg-slate-200"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            .
          </button>
          <button
            class="bg-magic-blue/80 text-white"
            data-mdb-ripple="true"
            data-mdb-ripple-color="light"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
